OrbitQt/OverlayWidget.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_QT_OVERLAY_WIDGET_H_
+#define ORBIT_QT_OVERLAY_WIDGET_H_
+
+#include <QEvent>
+#include <QObject>
+#include <QPainter>
+#include <QWidget>
+#include <memory>
+
+#include "OrbitBase/Logging.h"
+#include "ui_OverlayWidget.h"
+
+namespace OrbitQt {
+
+class OverlayWidget : public QWidget {
+  Q_OBJECT
+  Q_PROPERTY(bool spinning READ IsSpinning WRITE SetSpinning)
+  Q_PROPERTY(bool cancelable READ IsCancelable WRITE SetCancelable);
+  Q_PROPERTY(QString statusMessage READ GetStatusMessage WRITE SetStatusMessage)
+  Q_PROPERTY(QString buttonMessage READ GetButtonMessage WRITE SetButtonMessage);
+
+ public:
+  explicit OverlayWidget(QWidget* parent);
+
+ protected:
+  void paintEvent(QPaintEvent* /*event*/) override;
+  bool eventFilter(QObject* obj, QEvent* event) override;
+
+ private slots:
+  void on_cancelButton_clicked() { emit Cancelled(); }
```

Comment:
Personal preference: I don't like slots connected by name because the compiler can't check them. You will only notice at runtime that something changed. Maybe the name of "cancelButton". I would prefer to use QObject::connect and a lambda in the constructor of OverlayWidget, especially since the lambda will contain "emit Cancelled();".

Reply:
But we have these in several places in the codebase, so feel free to leave it here as well.

Reply:
No, your argumentation is very sound. I agree. 

I was wondering if there is a better way to accomplish what I've done here. Its basically just re-emiting a different signal. I guess `QObject::connect()` is the best way.

Reply:
done

Reply:
Yes, re-emitting is the best we can do here! Thanks!